src/main/java/edu/ie3/datamodel/utils/validation/UniquenessValidationUtils.java

```diff
@@ -0,0 +1,177 @@
+/*
+ * © 2024. TU Dortmund University,
+ * Institute of Energy Systems, Energy Efficiency and Energy Economics,
+ * Research group Distribution grid planning and operation
+*/
+package edu.ie3.datamodel.utils.validation;
+
+import edu.ie3.datamodel.exceptions.DuplicateEntitiesException;
+import edu.ie3.datamodel.io.source.TimeSeriesMappingSource.MappingEntry;
+import edu.ie3.datamodel.models.Entity;
+import edu.ie3.datamodel.models.UniqueEntity;
+import edu.ie3.datamodel.models.input.AssetInput;
+import edu.ie3.datamodel.models.result.ResultEntity;
+import edu.ie3.datamodel.models.timeseries.individual.TimeBasedValue;
+import edu.ie3.datamodel.models.value.WeatherValue;
+import edu.ie3.datamodel.utils.Try;
+import edu.ie3.datamodel.utils.Try.*;
+import java.util.*;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+/** Validation utils for checking the uniqueness of a given collection of entities. */
+public class UniquenessValidationUtils extends ValidationUtils {
+
+  // default field set supplier
+  protected static final FieldSetSupplier<UniqueEntity> uuidFieldSupplier =
+      entity -> Set.of(entity.getUuid());
+  protected static final FieldSetSupplier<AssetInput> idFieldSupplier = e -> Set.of(e.getId());
+  protected static final FieldSetSupplier<ResultEntity> resultFieldSupplier =
+      entity -> Set.of(entity.getTime(), entity.getInputModel());
+  protected static final FieldSetSupplier<MappingEntry> mappingFieldSupplier =
+      entity -> Set.of(entity.participant());
+  protected static final FieldSetSupplier<TimeBasedValue<WeatherValue>>
+      timeBasedValueFieldSupplier =
```

Comment:
One more point: This should probably be `weatherValueFieldSupplier`, right?

Reply:
Yes, that is a better name.